docs/USAGE-GUIDE.md

```diff
@@ -495,6 +495,20 @@ To use OCSP stapling, both server and client must call `s2n_config_set_status_re
 
 The OCSP stapling information will be automatically validated if the underlying libcrypto supports OCSP validation. `s2n_config_set_check_stapled_ocsp_response()` can be called with "0" to turn this off. Call `s2n_connection_get_ocsp_response()` to retrieve the received OCSP stapling information for manual verification.
 
+### CRL Validation
+
+Certificate Revocation Lists (CRLs) are lists of issued, unexpired certificates that have been revoked by the CA. CAs publish updated versions of these lists periodically. A peer wishing to verify a certificate can obtain a CRL from the CA, validate the CRL to make sure it was issued by the CA, and check to ensure the given certificate’s serial number is not contained in the list, and thus has not been revoked by the CA.
+
+To enable CRL validation in s2n-tls, the s2n CRL lookup callback must be implemented via `s2n_config_set_crl_lookup_callback()`. This callback is triggered once for each certificate received in the handshake. The received certificate's issuer hash can be retrieved in the callback via `s2n_crl_lookup_get_cert_issuer_hash()`. The issuer hash can be used to find the correct CRL for the given certificate.
+
+The s2n CRL lookup callback expects a CRL to be returned via `s2n_crl_lookup_set()`. `s2n_crl`s can be created from PEMs via `s2n_crl_new()` and `s2n_crl_load_pem()`, and then freed via `s2n_crl_free()`. The issuer hash of the CRL can be obtained with `s2n_crl_get_issuer_hash()`. This issuer hash of a certificate will match the issuer hash of its CRL.
```

Comment:
```suggestion
Create a `s2n_crl` by calling `s2n_crl_new()` and add the obtained CRL data to the `s2n_crl` by calling `s2n_crl_load_pem()`.  Then call `s2n_crl_lookup_set()` to associate the CRL with a specific certificate in the callback. The `s2n_crl` should be freed via `s2n_crl_free()` when no longer needed. 

Call `s2n_crl_get_issuer_hash()` to get the issuer hash of the CRL if necessary. 
```

Not sure if I love my edit, but this is probably how I would write this paragraph. I switched the order of your instructions because they should be the order the user would call these functions(unless I'm mistaken).

I would also add this paragraph to the previous one as it is explaining what to do with the CRL data obtained.

Reply:
Switching the ordering definitely makes sense. I added some stuff about what the application should do here, and tried to make the ordering more correct.